filecoin-proofs/src/api/update.rs

```diff
@@ -388,11 +430,14 @@ pub fn generate_partition_proofs<Tree: 'static + MerkleTreeTrait<Hasher = TreeRH
     sector_key_cache_path: &Path,
     replica_path: &Path,
     replica_cache_path: &Path,
-) -> Result<Vec<PartitionProof<Tree>>> {
+) -> Result<Vec<PartitionProof<Tree::Arity, Tree::SubTreeArity, Tree::TopTreeArity>>>
```

Comment:
Note: `PartitionProof` is no longer generic over `TreeR: MerkleTreeTrait`; it is now generic over `U, V, W`.